src/lib/mina_ledger/root.ml

```diff
@@ -60,19 +60,72 @@ module Make
                             and type converting_ledger = Unstable_db.t) =
 struct
   module Config = struct
-    type t = string
-
-    type backing_type = Stable_db
-
-    let exists_backing = Sys.file_exists
-
-    let with_directory ~backing_type:Stable_db ~directory_name = directory_name
-
-    let delete_any_backing = Mina_stdlib_unix.File_system.rmrf
-
-    let move_backing_exn ~src ~dst = Sys.rename src dst
-
-    let primary_directory = Fn.id
+    type backing_type = Stable_db | Converting_db [@@deriving equal]
+
+    type t =
+      | Stable_db_config of string
+      | Converting_db_config of Converting_ledger.Config.t
+
+    let backing_of_config = function
+      | Stable_db_config _ ->
+          Stable_db
+      | Converting_db_config _ ->
+          Converting_db
+
+    let file_exists path =
+      Sys.file_exists path |> [%equal: [ `No | `Unknown | `Yes ]] `Yes
+
+    let exists_backing = function
+      | Stable_db_config path ->
+          file_exists path
+      | Converting_db_config { primary_directory; converting_directory } ->
+          file_exists primary_directory && file_exists converting_directory
+
+    (* TODO: we should be able to tell backing type of a root just by looking at
+       the dir structure, maybe it can be utilized to simplify our code *)
+    let with_directory ~backing_type ~directory_name =
+      match backing_type with
+      | Stable_db ->
+          Stable_db_config directory_name
+      | Converting_db ->
+          Converting_db_config
+            (Converting_ledger.Config.with_primary ~directory_name)
+
+    let delete_any_backing = function
```

Comment:
The last thing is that I was considering having this function delete both kinds of backing, so that if a node happened to be started up in auto mode and then legacy mode then the converting database would still be cleaned up when this function is used.

Reply:
Now there's 2 version, `delete_backing` and `delete_any_backing`. Use the later when we don't know the backing. Now it seems only the previous function is used because all call site provides a config instead of directory.

Reply:
Yes, I think we'll do that in follow up PRs. 
> The issue is that the config we'll be getting in some of those places will be the current node's config, but we'll want to clean up the backing from old instances of the node.

Make sense, but I think it's better to consider replacing them when there's a HF cli flag around. I expect that would affect related logic.